### PR TITLE
Mocking database examples on the guide

### LINKED ISSUE #2871: Guide for database skipped
Within `docs/guide/acquiring_data/database.rst`, are many database examples. Recently we have to skip the entire file due to one big issue. 

The results of the online file retrieval varies on the source of the test or even the day. Basically if any of the online servers either report different data or are down, the file will fail. This means that all the large text need to be updated and it's very tedious work. 

We need a better way but right now, we don't have one.

### PR DESCRIPTION
**Pull request by dpshelio**
This should be able to generate the documentation without breaking the tests. Completely offline! if we want to check that the results still apply, then we can remove the two `testsetup` sections and it should work. How to do that in an automated way... I don't know.

Fixes #2871 - I think...

Also improved the way database looks into a directory by sorting the files within it, so we get always the same result.

### CODE REVIEW

**Comment by nabobalis** [comment]
If this works. You are a beautiful human being.

**Comment by dpshelio** [comment]
Why I don't get giles in here???

**Comment by dpshelio** [comment]
Ok, maybe because the last commit said `[ci skip]`. I think [this is the built documentation](https://5154-2165383-gh.circle-artifacts.com/0/root/project/docs/_build/html/guide/acquiring_data/database.html).

**Comment by nabobalis** [comment]
```
2018-12-16T20:29:25.1945060Z _________________________________________________________________ [doctest] docs/guide/acquiring_data/database.rst _________________________________________________________________
2018-12-16T20:29:25.1945480Z 251 .. testsetup:: *
2018-12-16T20:29:25.1945790Z 252    >>> from unittest.mock import Mock
2018-12-16T20:29:25.1946140Z 253    >>> from sunpy.tests.mocks import MockObject
2018-12-16T20:29:25.1946660Z 254    >>> class MockQR:
2018-12-16T20:29:25.1946950Z 255    ...    def __init__(self, start_time, instrument, wavelength):
2018-12-16T20:29:25.1948510Z 256    ...        end_time = '{}{}'.format(start_time[:-1], int(start_time[-1]) + 1)
2018-12-16T20:29:25.1948650Z 257    ...        self.time = MockObject(start=start_time, end=end_time)
2018-12-16T20:29:25.1948700Z 258    ...        self.instrument = instrument
2018-12-16T20:29:25.1949790Z 259    ...        self.wave = MockObject(wavemin=wavelength, wavemax=wavelength, waveunit='nm')
2018-12-16T20:29:25.1949920Z 260    >>> client.search = Mock(return_value=[MockQR(f"2011050800000{t}",
2018-12-16T20:29:25.1952980Z UNEXPECTED EXCEPTION: NameError("name 'client' is not defined")
2018-12-16T20:29:25.1954020Z Traceback (most recent call last):
2018-12-16T20:29:25.1954110Z 
2018-12-16T20:29:25.1954880Z   File "/Users/vsts/agent/2.144.0/work/1/s/.tox/py37-offline/lib/python3.7/doctest.py", line 1329, in __run
2018-12-16T20:29:25.1955020Z     compileflags, 1), test.globs)
2018-12-16T20:29:25.1955050Z 
2018-12-16T20:29:25.1955140Z   File "<doctest database.rst[23]>", line 3, in <module>
2018-12-16T20:29:25.1955190Z 
2018-12-16T20:29:25.1955680Z NameError: name 'client' is not defined
2018-12-16T20:29:25.1955730Z 
2018-12-16T20:29:25.1955780Z /Users/vsts/agent/2.144.0/work/1/s/docs/guide/acquiring_data/database.rst:260: UnexpectedException2018-12-16T20:29:25.1945060Z _________________________________________________________________ [doctest] docs/guide/acquiring_data/database.rst _________________________________________________________________
2018-12-16T20:29:25.1945480Z 251 .. testsetup:: *
2018-12-16T20:29:25.1945790Z 252    >>> from unittest.mock import Mock
2018-12-16T20:29:25.1946140Z 253    >>> from sunpy.tests.mocks import MockObject
2018-12-16T20:29:25.1946660Z 254    >>> class MockQR:
2018-12-16T20:29:25.1946950Z 255    ...    def __init__(self, start_time, instrument, wavelength):
2018-12-16T20:29:25.1948510Z 256    ...        end_time = '{}{}'.format(start_time[:-1], int(start_time[-1]) + 1)
2018-12-16T20:29:25.1948650Z 257    ...        self.time = MockObject(start=start_time, end=end_time)
2018-12-16T20:29:25.1948700Z 258    ...        self.instrument = instrument
2018-12-16T20:29:25.1949790Z 259    ...        self.wave = MockObject(wavemin=wavelength, wavemax=wavelength, waveunit='nm')
2018-12-16T20:29:25.1949920Z 260    >>> client.search = Mock(return_value=[MockQR(f"2011050800000{t}",
2018-12-16T20:29:25.1952980Z UNEXPECTED EXCEPTION: NameError("name 'client' is not defined")
2018-12-16T20:29:25.1954020Z Traceback (most recent call last):
2018-12-16T20:29:25.1954110Z 
2018-12-16T20:29:25.1954880Z   File "/Users/vsts/agent/2.144.0/work/1/s/.tox/py37-offline/lib/python3.7/doctest.py", line 1329, in __run
2018-12-16T20:29:25.1955020Z     compileflags, 1), test.globs)
2018-12-16T20:29:25.1955050Z 
2018-12-16T20:29:25.1955140Z   File "<doctest database.rst[23]>", line 3, in <module>
2018-12-16T20:29:25.1955190Z 
2018-12-16T20:29:25.1955680Z NameError: name 'client' is not defined
2018-12-16T20:29:25.1955730Z 
2018-12-16T20:29:25.1955780Z /Users/vsts/agent/2.144.0/work/1/s/docs/guide/acquiring_data/database.rst:260: UnexpectedException
```

**Comment by Cadair** [comment]
if this is now all offline we can and should go through and remove all the `REMOTE_DATA` directives.

**Comment by dpshelio** [comment]
testsetup blocks are supposed to be in the same namespace. That's the idea of the testsetup, at least with a `*` on them. I don't really know why that doctest here. Ideally (and I don't know how we could do this) we could name these testsetups so that we could omit them if we want to actually update the outputs as using the online data.

**Comment by Cadair** [comment]
maybe they are always run first? Either way I think you just need to patch `VSOClient.search` directly rather than doing it on the `client` variable.

**Comment by nabobalis** [comment]
My fix was not a fix. Back to the drawing board.

File: sunpy/database/tables.py [hunk 1]
```diff
@@ -645,6 +645,14 @@ def entries_from_file(file, default_waveunit=None,
...
     """
     headers = fits.get_header(file)
+    headers = fits.get_header(file)
+    # This is due to RICE compression adds a second header
+    # with no information in it. If we do not remove it here,
+    # it gets added to the database.
+    for header in headers:
+        if header['NAXIS'] == 0 and len(header) == 8:
+            headers.remove(header)
+
```
Comment (nabobalis) [inline sunpy/database/tables.py]: On a scale of 1 to hack, this is a hack.

File: .circleci/config.yml [hunk 2]
```diff
@@ -60,7 +60,6 @@ jobs:
 
       - checkout
       - run: *skip-check
-      - run: *merge-check
```
Comment (nabobalis) [inline .circleci/config.yml]: Running ssh on such an old system is a bad idea for my health.

**Comment by nabobalis** [comment]
App fail is fixed in another PR.
The 32bit one is also fixed in another PR. 

The rest should pass!

**Comment by Cadair** [comment]
I would like to see at least one test build pass :stuck_out_tongue: Looks like we have issues with the latest pytest again.

**Comment by nabobalis** [comment]
Rebased it on a working CI. Let us see what happens!

**Comment by nabobalis** [comment]
I restarted the build but it seems to have been fixed now.

**Comment by nabobalis** [comment]
ALL GREEN

**Pull request merged**